packages/clerk-js/src/core/resources/Environment.ts

```diff
@@ -54,9 +53,9 @@ export class Environment extends BaseResource implements EnvironmentResource {
     this.clientDebugMode = this.withDefault(data.client_debug_mode, this.clientDebugMode);
     this.organizationSettings = new OrganizationSettings(data.organization_settings);
     this.userSettings = new UserSettings(data.user_settings);
-    this.commerceSettings = new CommerceSettings(data.commerce_settings);
-    this.apiKeysSettings = new APIKeySettings(data.api_keys_settings);
-    this.protectConfig = new ProtectConfig(data.protect_config);
+    this.commerceSettings = new CommerceSettings(data.commerce_settings ?? null);
+    this.apiKeysSettings = new APIKeySettings(data.api_keys_settings ?? null);
```

Comment:
Looking at the backend code for these fields, I don't think they're potentially undefined. Did you run into that scenario when testing?